### PR TITLE
PLANNER-2795 Reduce CI verbosity

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/PLANNER-2795

- All failed builds (regardless of branch/job name/etc) go to the CI failures topic in optaplanner-dev.
- All success build (regardless of branch/job name/etc) go to the CI success topic in optaplanner-dev.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] tests</b>
  
- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: 
    please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: 
    add the label `run_fdb`

- for a <b>compile downstream build</b>  
  please add comment: <b>Jenkins run cdb</b>

- for a <b>full production downstream build</b>  
  please add comment: <b>Jenkins execute product fdb</b>

- for an <b>upstream build</b>  
  please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] mandrel</b>
</details>

### CI Status

 You can check OptaPlanner repositories CI status from [Chain Status webpage](https://kiegroup.github.io/optaplanner/).

### CODE REVIEW

**Comment by radtriste** [comment]
@ge0ffrey @rsynek For now I just updated current jobs. Most of them are only reporting in case of failures. Do we want them also to notify in case of success ?

File: .ci/jenkins/Jenkinsfile.deploy [hunk 1]
```diff
@@ -258,9 +258,10 @@ pipeline {
     }
 }
 
-void sendNotification() {
+void sendErrorNotification() {
     if (params.SEND_NOTIFICATION) {
-        mailer.sendMarkdownTestSummaryNotification('Deploy', "[${getBuildBranch()}] Optaplanner", [env.OPTAPLANNER_CI_EMAIL_TO])
+        String additionalInfo = "[${getBuildBranch()}] Optaplanner - Deploy"
```
Comment (rsynek) [inline .ci/jenkins/Jenkinsfile.deploy:263]:
```suggestion
        String additionalInfo = "[${getBuildBranch()}] OptaPlanner - Deploy"
```

File: .ci/jenkins/Jenkinsfile.init-branch [hunk 2]
```diff
@@ -94,9 +94,10 @@ pipeline {
     }
 }
 
-void sendNotification() {
+void sendErrorNotification() {
     if (params.SEND_NOTIFICATION) {
-        mailer.sendMarkdownTestSummaryNotification('Init', "[${getBuildBranch()}] Optaplanner", [env.OPTAPLANNER_CI_EMAIL_TO])
+        String additionalInfo = "[${getBuildBranch()}] Optaplanner - Init"
```
Comment (rsynek) [inline .ci/jenkins/Jenkinsfile.init-branch:99]:
```suggestion
        String additionalInfo = "[${getBuildBranch()}] OptaPlanner - Init"
```

File: .ci/jenkins/Jenkinsfile.promote [hunk 3]
```diff
@@ -214,9 +214,10 @@ pipeline {
     }
 }
 
-void sendNotification() {
+void sendErrorNotification() {
     if (params.SEND_NOTIFICATION) {
-        mailer.sendMarkdownTestSummaryNotification('Promote', "[${getBuildBranch()}] Optaplanner", [env.OPTAPLANNER_CI_EMAIL_TO])
+        String additionalInfo = "[${getBuildBranch()}] Optaplanner - Promote"
```
Comment (rsynek) [inline .ci/jenkins/Jenkinsfile.promote:219]:
```suggestion
        String additionalInfo = "[${getBuildBranch()}] OptaPlanner - Promote"
```

File: .ci/jenkins/Jenkinsfile.turtle [hunk 4]
```diff
@@ -81,8 +86,17 @@ String getConstraintStreamImplType() {
...
+    mailer.sendMarkdownTestSummaryNotification("CI failures", [env.OPTAPLANNER_CI_EMAIL_TO], getNotificationAdditionalInfo())
+}
+
+void sendErrorNotification() {
+    mailer.sendMarkdownTestSummaryNotification("CI success", [env.OPTAPLANNER_CI_EMAIL_TO], getNotificationAdditionalInfo())
+}
+
+String getNotificationAdditionalInfo() {
+    return """
+    [${getBuildBranch()}] Optaplanner - Turtle tests
```
Comment (rsynek) [inline .ci/jenkins/Jenkinsfile.turtle:99]:
```suggestion
    [${getBuildBranch()}] OptaPlanner - Turtle tests
```

File: .ci/jenkins/project/Jenkinsfile.drools [hunk 5]
```diff
@@ -119,8 +119,12 @@ pipeline {
     }
 }
 
-void sendNotification() {
-    mailer.sendMarkdownTestSummaryNotification("${NOTIFICATION_JOB_NAME}", "[${getBuildBranch()}] Optaplanner", [env.OPTAPLANNER_CI_EMAIL_TO], "cc @*drools-team*")
+void sendErrorNotification() {
+    String additionalInfo = """
+    [${getBuildBranch()}] Optaplanner${NOTIFICATION_JOB_NAME ? " - ${NOTIFICATION_JOB_NAME}" : ''}
```
Comment (rsynek) [inline .ci/jenkins/project/Jenkinsfile.drools:124]:
```suggestion
    [${getBuildBranch()}] OptaPlanner${NOTIFICATION_JOB_NAME ? " - ${NOTIFICATION_JOB_NAME}" : ''}
```

File: .ci/jenkins/project/Jenkinsfile.native [hunk 6]
```diff
@@ -81,8 +81,10 @@ pipeline {
     }
 }
 
-void sendNotification() {
-    mailer.sendMarkdownTestSummaryNotification("${NOTIFICATION_JOB_NAME}", "[${params.BUILD_BRANCH_NAME}] Optaplanner", [env.OPTAPLANNER_CI_EMAIL_TO])
+
+void sendErrorNotification() {
+    String additionalInfo = "[${getBuildBranch()}] Optaplanner${NOTIFICATION_JOB_NAME ? " - ${NOTIFICATION_JOB_NAME}" : ''}"
```
Comment (rsynek) [inline .ci/jenkins/project/Jenkinsfile.native:86]:
```suggestion
    String additionalInfo = "[${getBuildBranch()}] OptaPlanner${NOTIFICATION_JOB_NAME ? " - ${NOTIFICATION_JOB_NAME}" : ''}"
```

File: .ci/jenkins/project/Jenkinsfile.quarkus [hunk 7]
```diff
@@ -102,8 +102,9 @@ pipeline {
     }
 }
 
-void sendNotification() {
-    mailer.sendMarkdownTestSummaryNotification("Quarkus ${getQuarkusBranch()}", "[${getBuildBranch()}] Optaplanner", [env.OPTAPLANNER_CI_EMAIL_TO])
+void sendErrorNotification() {
+    String additionalInfo = "[${getBuildBranch()}] Optaplanner - Quarkus ${getQuarkusBranch()}}"
```
Comment (rsynek) [inline .ci/jenkins/project/Jenkinsfile.quarkus:106]:
```suggestion
    String additionalInfo = "[${getBuildBranch()}] OptaPlanner - Quarkus ${getQuarkusBranch()}}"
```

**Review by rsynek — APPROVED** [review]
I think this notification level is ok; we get a success notification from turtle tests and a failure notification from everything.
Not sure how much the nightly and release pipelines notify now; it would be nice to at least know they started and finished (no details about their stages needed).

**Comment by radtriste** [comment]
> I think this notification level is ok; we get a success notification from turtle tests and a failure notification from everything. Not sure how much the nightly and release pipelines notify now; it would be nice to at least know they started and finished (no details about their stages needed).

I did not change the overall notification of nightly/release, just removed the notification if the job is UNSTABLE as we get that info from the different called jobs already. 
As of now, the nightly will only notify is we get an error in the pipeline itself. The release pipeline does already notify when starting/ending and when each deploy/promote stages are done.

**Comment by radtriste** [comment]
@rsynek all good for merging (if you want to have a final look)

File: .ci/jenkins/Jenkinsfile.turtle [hunk 8]
```diff
@@ -81,8 +86,17 @@ String getConstraintStreamImplType() {
...
+}
+
+void sendErrorNotification() {
+    mailer.sendMarkdownTestSummaryNotification("CI success", [env.OPTAPLANNER_CI_EMAIL_TO], getNotificationAdditionalInfo())
+}
+
+String getNotificationAdditionalInfo() {
+    return """
+    **[${getBuildBranch()}] Optaplanner - Turtle tests**
+    **"${getConstraintStreamImplType()}"**
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.turtle:100]:
```suggestion
    **${getConstraintStreamImplType()}**
```

**Pull request merged**